jwst/ramp_fitting/tests/test_ramp_fit.py

```diff
@@ -0,0 +1,168 @@
+    print(slopes[0].data[500,500])
+    # take the ratio of the slopes to get the relative error
+    assert_almost_equals(slopes[0].data[500, 500]/ (20.0/3), 1.0, places=6)
+
+#@pytest.mark.skip(reason="not using now")
+def test_read_noise_only_fit():
+        #
+    model1, gdq, rnModel, pixdq, err, gain = setup_inputs(ngroups=5,readnoise=50)
+    model1.data[0, 0, 500, 500] = 10.0
+    model1.data[0, 1, 500, 500] = 15.0
+    model1.data[0, 2, 500, 500] = 25.0
+    model1.data[0, 3, 500, 500] = 33.0
+    model1.data[0, 4, 500, 500] = 60.0
+    slopes = ramp_fit(model1, 64000, True, rnModel, gain, 'OLS', 'optimal')
+    xvalues = np.arange(5)*1.0
+    yvalues = np.array([10,15,25,33,60])
+    coeff = np.polyfit(xvalues, yvalues, 1)
+    print('x ',repr(xvalues))
+    print('coeff ',repr(coeff))
+    assert_almost_equals(slopes[0].data[500, 500]/coeff[0], 1.0, places=6)
+
+#@pytest.mark.skip(reason="not using now")
+def test_photon_noise_only_fit():
+        #
+    model1, gdq, rnModel, pixdq, err, gain = setup_inputs(ngroups=5,gain=1000,readnoise=1)
+    model1.data[0, 0, 500, 500] = 10.0
+    model1.data[0, 1, 500, 500] = 15.0
+    model1.data[0, 2, 500, 500] = 25.0
+    model1.data[0, 3, 500, 500] = 33.0
+    model1.data[0, 4, 500, 500] = 60.0
+    slopes = ramp_fit(model1, 64000, True, rnModel, gain, 'OLS', 'optimal')
+    cds_slope = (model1.data[0,4,500,500] - model1.data[0,0,500,500])/ 4.0
+    print('CDS Slope', cds_slope)
+    print('Slope ',slopes[0].data[500, 500])
+
+    assert_almost_equals(slopes[0].data[500, 500]/cds_slope, 1.0, places=2)
+
+#@pytest.mark.skip(reason="not using now")
+def test_two_groups_fit():
+        #
+    model1, gdq, rnModel, pixdq, err, gain = setup_inputs(ngroups=2,gain=1,readnoise=10)
+    model1.data[0, 0, 500, 500] = 10.0
+    model1.data[0, 1, 500, 500] = 15.0
+    slopes = ramp_fit(model1, 64000, True, rnModel, gain, 'OLS', 'optimal')
+    cds_slope = (model1.data[0,1,500,500] - model1.data[0,0,500,500])
+    print('CDS Slope', cds_slope)
+    print('Slope ',slopes[0].data[500, 500])
+
+    assert_almost_equals(slopes[0].data[500, 500]/cds_slope, 1.0, places=2)
+
+#ramp_fit_step hardcodes the input to be OLS. So you can't get to the GLS code.
+@pytest.mark.skip(reason="Fails")
+def test_simple_gls_ramp():
+    #Here given a 10 group ramp with an exact slope of 20/group. The output slope should be 20.
+    model1, gdq, rnModel, pixdq, err, gain = setup_inputs(ngroups=10)
+    model1.data[0, 0, 500, 500] = 10.0
+    model1.data[0, 1, 500, 500] = 30.0
+    model1.data[0, 2, 500, 500] = 50.0
+    model1.data[0, 3, 500, 500] = 70.0
+    model1.data[0, 4, 500, 500] = 90.0
+    model1.data[0, 5, 500, 500] = 110.0
+    model1.data[0, 6, 500, 500] = 130.0
+    model1.data[0, 7, 500, 500] = 150.0
+    model1.data[0, 8, 500, 500] = 170.0
+    model1.data[0, 9, 500, 500] = 190.0
+    slopes = ramp_fit(model1, 64000, True, rnModel, gain, 'GLS', 'optimal')
```

Comment:
The GLS option isn't supported.  I don't have any objection to this test; I'm just a little surprised that it still works.